packages/contracts-bedrock/src/dispute/FaultDisputeGame.sol

```diff
@@ -552,10 +554,47 @@ contract FaultDisputeGame is IFaultDisputeGame, Clone, ISemver {
     /// @notice Returns the required bond for a given move kind.
     /// @param _position The position of the bonded interaction.
     /// @return requiredBond_ The required ETH bond for the given move, in wei.
-    function getRequiredBond(Position _position) public pure returns (uint256 requiredBond_) {
-        // TODO(client-pod#551): For now use a non-zero bond amount to unblock functional tests.
-        _position;
-        requiredBond_ = 0.01 ether;
+    function getRequiredBond(Position _position) public view returns (uint256 requiredBond_) {
+        uint256 depth = uint256(_position.depth());
+        if (depth > MAX_GAME_DEPTH) revert GameDepthExceeded();
+
+        // Values taken from Big Bonds v1.5 (TM) spec.
+        uint256 assumedBaseFee = 200 gwei;
+        uint256 baseGasCharged = 400_000;
+        uint256 highGasCharged = 200_000_000;
+
+        // Goal here is to compute the fixed multiplier that will be applied to the base gas
+        // charged to get the required gas amount for the given depth. We apply this multiplier
+        // some `n` times where `n` is the depth of the position. We are looking for some number
+        // that, when multiplied by itself `MAX_GAME_DEPTH` times and then multiplied by the base
+        // gas charged, will give us the maximum gas that we want to charge.
+        // We want to solve for (highGasCharged/baseGasCharged) ** (1/MAX_GAME_DEPTH).
+        // We know that a ** (b/c) is equal to e ** (ln(a) * (b/c)).
+        // We can compute e ** (ln(a) * (b/c)) quite easily with FixedPointMathLib.
+
+        // Set up a, b, and c.
+        uint256 a = highGasCharged / baseGasCharged;
+        uint256 b = FixedPointMathLib.WAD;
+        uint256 c = MAX_GAME_DEPTH * FixedPointMathLib.WAD;
+
+        // Compute ln(a).
+        // slither-disable-next-line divide-before-multiply
+        uint256 lnA = uint256(FixedPointMathLib.lnWad(int256(a * FixedPointMathLib.WAD)));
+
+        // Computes (b / c) with full precision using WAD = 1e18.
+        uint256 bOverC = FixedPointMathLib.divWad(b, c);
+
+        // Compute e ** (ln(a) * (b/c))
+        // sMulWad can be used here since WAD = 1e18 maintains the same precision.
+        uint256 numerator = FixedPointMathLib.mulWad(lnA, bOverC);
+        int256 base = FixedPointMathLib.expWad(int256(numerator));
+
+        // Compute the required gas amount.
+        int256 rawGas = FixedPointMathLib.powWad(base, int256(depth * FixedPointMathLib.WAD));
+        uint256 requiredGas = FixedPointMathLib.mulWad(baseGasCharged, uint256(rawGas));
+
+        // Compute the required bond.
+        requiredBond_ = assumedBaseFee * requiredGas;
```

Comment:
So far we cover at least the gas cost of countering claims, given the `assumedBaseFee`. Is the idea here that the `assumedBaseFee` not only covers increases in the base fee, but also, for example, offchain costs?

Basically, I'd like us to be explicit whether we want to ensure participation in the case where `basefee == assumedBaseFee`. With a high enough `assumedBaseFee`, the final bond calculated will effectively cover all other costs (including opportunity costs). This is particularly important if we decide to let governance control `assumedBaseFee` and the purpose of that value is lost.

Reply:
Operating a challenger also provides opportunity though in it of itself in the form of seizing an opponent's bonds in the case where your challenger is honest and there are opponents posting invalid outputs. What I mean to say is it's not entirely satisfactory to say `assumedBaseFee` should cover opportunity cost. It might be more accurate to implement a mechanism that drips incentives where no dishonest or invalid outputs are posted. WRT this pr though, would it make sense to spec out these incentives first then follow on with a separate implementation?

Reply:
An opportunity cost always exists. If bond rewards == gas costs, then unless one is altruistic and likes Optimism, there isn't an incentive to participate. Which could be a fair assumption to have, but one we should call out.

I believe the specification used to mention all of these off-protocol incentives, but now I can't find it. We shouldn't add these costs to the bond function in this PR. I'm only saying that we should document what other costs are being assumed away (or not). But perhaps we should clarify this in the specs instead.